Slim/Control/Queries.pm

```diff
@@ -644,9 +648,9 @@ sub albumsQuery {
 	$sql .= "ORDER BY $order_by " unless $tags eq 'CC';
 
 	# Add selected columns
-	# Bug 15997, AS mapping needed for MySQL
+	# Bug 15997, AS mapping needed for MySQL ** But not if the column is a subselect that already has an 'AS' provided **
 	my @cols = sort keys %{$c};
-	$sql = sprintf $sql, join( ', ', map { $_ . " AS '" . $_ . "'" } @cols );
+	$sql = sprintf $sql, join( ', ', map { $_ . ($_ =~ /SELECT.*AS/ ? "" : " AS '" . $_ . "'") } @cols );
```

Comment:
Whenever I see the same obscure code being implemented twice or more, I feel an urge to factor it out into its own `sub`. Please?